templates/pre-commit-hook.js

```diff
@@ -0,0 +1,32 @@
+ // hooks/pre-commit.js
+
+ const exec = require('child_process').exec;
+ // Executes shell commands synchronously
+ const sh = require('child_process').execSync;
+ 
+ exec('git diff --cached --quiet', function (err, stdout, stderr) {
+ 
+  // only run if there are staged changes
+  // i.e. what you would be committing if you ran "git commit" without "-a" option.
+  if (err) {
```

Comment:
@frappelatte28  i am still wondering why `err` has to be there , should not it be `!err`  ?

Reply:
yes it is interesting to note that , we are using `--quiet` flag here that reverses the error code to `1` if there are stages changes and `0` if not. 
we can remove `--quiet` and use things as normal.you can look at the reference https://stackoverflow.com/a/28296286

Reply:
Ohh great ! I think that was the concern and it's to get it on board.